File: snippets/MethodOverloadingArity.exs
defmodule MethodOverloadingArity do

    def join(a, b) do
        join(a, b, " ")
    end

    def join(a, b, sep) do
        a <> sep <> b  # string concatenation
    end

end
# enddefmodule
foo = MethodOverloadingArity.join
foo.("a", "b")
foo.("a", "b", " - ")

File: snippets/MethodOverloadingArity.java
public class MethodOverloadingArity {

    public static String join(String a, String b) {
        return join(a, b, " ");
    }

    public static String join(String a, String b, String sep) {
        return String.format("%s%s%s", a, sep, b);
    }
}
